plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java

```diff
@@ -814,7 +811,7 @@ private Map<String, Map<String, Map<String, String>>> readSecureCredentials(
                         log.error("Error backing up secure credentials", e1);
                     }
                 } else {
-                    throw new DBException("Project opening canceled by user");
+                    log.error("Project opening canceled by user");
```

Comment:
This **won't** prevent the project from opening. The main idea was to throw an exception so the node wouldn't be fully loaded. See `DBNProject#readChildNodes`.

Reply:
In result, user able to operate with unsecured projects as before, the secured assets present in tree but is not available for use.

Reply:
This is current behavior (before this ticket). The idea was to completely block the user from opening the project because as soon they make changes to it (change connection configuration, etc), it will be persisted and all credentials will be lost.

My approach was completely valid, the only issue that stopped my PR from being merged is that it behaves weirdly on macOS. This is what needs to be fixed.

Reply:
No objection of your approach, it's valid, the main issue on both OS's (Win Mac) is hitting "Cancel" button in the dialog.
Pls check the snapshort on win os

Uploading Recording 2024-01-17 122418.mp4…

Reply:
No recording is attached.

Reply:
It happens due to 
![Uploading Screenshot 2024-01-17 131156.png…]()

Reply:
Uploading Recording 2024-01-17 122418.mp4…

Reply:
Still no recording attached.